### ISSUE TITLE
batch controls to downloads

### ISSUE THREAD

**Issue by Mike-Reddick** [jingyu9575/multithreaded-download-manager#110]
I want ask for an feature to select multiple files to play, pause, stop, deleted, only selected files, and also an PLAY ALL/PAUSE ALL button on toolbar and an setting to PLAY ALL on browser startup. When all download are paused extension can even show the number of paused downloads with yellow badge. 

Thanks so much

**Comment by jingyu9575** [comment]
It is already suggested in #73. I think selection by mouse is not difficult, but making keyboard navigations work is much harder. Maybe I'll try the mouse selection first.

> When all download are paused extension can even show the number of paused downloads with yellow badge

I didn't put this to avoid inconsistency. When there is 2 files downloading and 1 paused, what should the badge color and text be?

**Comment by Mike-Reddick** [comment]
To me is fine the support for multi selection just using mouse without keyboard. About badge, so, if I have 100 downloads running, show the blue badge with 100, if have 100 paused downloads show the yellow badge with 100, if I have 80 running and 20 paused, show the blue badge with 80, in the end of this 80 downloads, if have some fails, show the red badge with fail counter, if haven't fails show the paused yellow badge. Have an badge priority:

1- Running
2- Fails
3- Paused

But is just suggestions, you know what is better to do, thanks

**Comment by jingyu9575** [comment]
Multi-selection with mouse is implemented:

![图片](https://user-images.githubusercontent.com/8923139/72533735-edca2b00-3875-11ea-97ce-19e9671cf8b8.png)

With the *select all* button it is possible to start or pause all tasks by 2 clicks.

**Comment by Mike-Reddick** [comment]
First, thanks so much for this update, is very fine, perfect multi selection, I want just make a little suggestion to review the dark theme on selected objects.

**Comment by jingyu9575** [comment]
> review the dark theme on selected objects

The background color has been changed.

---

> in the end of this 80 downloads, if have some fails, show the red badge with fail counter, if haven't fails show the paused yellow badge. Have an badge priority:
> 
> 1- Running
> 2- Fails
> 3- Paused

In this way, if there is 10 downloads and 1 suddenly failed, the badge will show 9 in blue, which is the same if 1 has completed. As a result, the user is not notified on the failure and only aware of it after the other 9 completes.

Currently the extension's badge color and number are separate. The number always show the running tasks, while the color notifies the user about completed/failed tasks.

**Comment by Mike-Reddick** [comment]
> In this way, if there is 10 downloads and 1 suddenly failed, the badge will show 9 in blue, which is the same if 1 has completed. As a result, the user is not notified on the failure and only aware of it after the other 9 completes.

Exactly, the behavior is not "perfect", as you said to know the fails only in the end, but by other side, if user "added paused" 100 downloads or paused it manually, and turned off pc and forget it, the user isn't notified about this paused downloads, sure? Then anyway the behavior is not absolutely perfect, an option can be an new setting, "enable advanced badges". But as I ever say, is just suggestions to you think about, is your hands. Thanks.

**Comment by Mike-Reddick** [comment]
The new multi selection is very good, I want just suggest some little things:

1- Add to toolbar Start button, Shift: Reset, (single or multi items)
2- Add to toolbar Remove button, Shift: Delete file (single or multi items)
3- Add an new button to export download list to an simple txt file as an url per line and Shift: Copy download Link (single or multi items)

This can be cool, just to you think about, some ideas...

**Comment by jingyu9575** [comment]
I'm temporarily suspending all feature requests because of a crucial change in Firefox 74. See #126.

**Comment by Mike-Reddick** [comment]
I hope mozilla don't frustrate us again.
